### PR TITLE
Internal: Implement atomic list element [ED-25001]

### PR DESCRIPTION
**Pull request by mike-elementor**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

## 1. Problem & Context

Implementing the atomic list element (ED-25001) to provide a semantic, structured list component with configurable markers and nested content slots. Enables marker visibility toggling via render context propagation from parent to children.

## 2. What Changed (Where)

| Category | Changes |
|----------|---------|
| **PHP Backend** | New element classes: `Atomic_List`, `Atomic_List_Item`, `Atomic_List_Item_Marker`, `Atomic_List_Item_Content`; control type `List_Items_Control`; module registration refactored into `register_list_element()` |
| **Frontend Canvas** | List-specific render context initialization in `list-type.ts`; type definitions extended for `initialize()` and `listenTo()` |
| **Editor Panel** | List items repeater control + actions (`addItem`, `removeItem`, `duplicateItem`, `moveItem`); write-through cascade for `show_markers` state |
| **Templates & Tests** | Twig templates for list/item/marker/content; PHPUnit + TypeScript + Playwright test coverage |

## 3. How It Works

List element establishes `show_markers` context via `define_render_context()` → propagates to each item child via `Render_Context::get(Atomic_List::class)` → `Child_Dependency` conditionally attaches/detaches marker slot based on prop value. Editor repeater mirrors root `show_markers` to items (undoable cascade) so item-level `Child_Dependency` triggers correctly. Markers stash when hidden, preserving customizations on re-enable. Default children: item gets content slot; content gets paragraph; marker gets SVG.

## 4. Risks

**Marker stashing complexity**: Conditional child rendering via dependencies + stashing introduces state management edge case if item state diverges from parent during rapid toggles—mitigated by undoable cascade batching and test coverage of undo/redo. **Edit panel architecture**: Write-through pattern for cascading state to items could clash if future prop changes bypass this path—document assumption or generalize pattern if needed.

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: packages/packages/core/editor-canvas/src/legacy/list-type.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+			this._syncShowMarkersToChildren();
+		},
+
+		/**
+		 * Sync show_markers from list to all list items.
+		 *
+		 * This propagates the list-level setting to each item, where children dependencies
+		 * will conditionally add/remove markers based on the value.
+		 */
+		_syncShowMarkersToChildren() {
```
Comment (Omerisra6) [inline packages/packages/core/editor-canvas/src/legacy/list-type.ts]: Try to delete it and check if it works

File: packages/packages/core/editor-canvas/src/init.tsx [hunk 2]
```diff
@@ -27,6 +28,9 @@ export function init() {
 
 	initViewReplacements();
 
+	initTabsModelExtensions();
```
Comment (Omerisra6) [inline packages/packages/core/editor-canvas/src/init.tsx]: Move both to the bottom of the file to keep the same loading order

File: packages/packages/core/editor-editing-panel/src/controls-registry/element-controls/list-items-control/list-actions.ts
Comment (Omerisra6) [inline packages/packages/core/editor-editing-panel/src/controls-registry/element-controls/list-items-control/list-actions.ts:1]: please check that everything works as expected for undo operations, for tabs we had issues with that

File: packages/packages/core/editor-editing-panel/src/controls-registry/element-controls/list-items-control/list-items-control.tsx [hunk 3]
```diff
@@ -29,8 +30,12 @@ const ListItemsControlContent = ( { label }: { label: string } ) => {
 		{ [ LIST_ELEMENT_TYPE ]: LIST_ITEM_ELEMENT_TYPE },
 		{ includeSelfAsParent: true }
 	);
+	// The root owns `show_markers`, but each child item needs a mirrored copy for its own
+	// `Child_Dependency`, so the repeater hosts the write-through that keeps those values undoable.
 	const showMarkers = booleanPropTypeUtil.extract( settings.show_markers ) ?? true;
 
+	useShowMarkersWriteThrough( element.id, showMarkers );
```
Comment (Omerisra6) [inline packages/packages/core/editor-editing-panel/src/controls-registry/element-controls/list-items-control/list-items-control.tsx:37]: check what is it needed for

Comment (mike-elementor) [reply packages/packages/core/editor-editing-panel/src/controls-registry/element-controls/list-items-control/list-items-control.tsx:37]: @Omerisra6 

So its job is:

keep root show_markers and item show_markers in sync
only react on real value flips, not on first mount or when selecting another list
write the updates as one undoable/debounced history action, so Undo restores the switch and all affected items together
This is the same pattern as accordion’s useShowIconWriteThrough(): root owns the setting, children carry mirrored copies because the dependency/reconciler logic runs against each child’s own settings, not the parent’s.

**Review by Omerisra6 — APPROVED** [review]
